README.md

```diff
@@ -1,176 +1,100 @@
-
-# More complex expressions
-iex> compare?(%{a: ~T[16:00:00]}.a <= ~T[17:00:00], Time)
-true
-```
+- chained comparisons (`a < b < c`)
+- semantic comparisons using structural operators (`<`, `>`, `<=`, `>=`, `==`, `!=`, `===`, and `!==`)
+- combinations (`and`, `or`, and `not`)
 
-## Installation
+## Background
 
-Add `compare_chain` to your list of dependencies in `mix.exs`:
-
-```elixir
-def deps do
-  [
-    {:compare_chain, "~> 0.5"}
-  ]
-end
-```
-
-Documentation can be found at <https://hexdocs.pm/compare_chain>.
-
-## Usage
-
-Once installed, you can add:
-
-```elixir
-import CompareChain
-```
-
-to your `defmodule` and you will have access to `CompareChain.compare?/1` and `CompareChain.compare?/2`.
-
-## Background and motivation
-
-`CompareChain` was originally motivated by the following situation:
-
-> You have an interval of time bounded by a two `%Date{}` structs: `start_date` and `end_date`.
-> You want to know if some third `date` falls in that interval.
-> How do you write this?
-
-In Elixir, we'd write this as follows:
+Say you have an interval of time bounded by two `%Date{}` structs, `start_date` and `end_date`, and want to know whether or not a third `date` falls within that interval. How would you write this in Elixir?
 
 ```elixir
 Date.compare(start_date, date) == :lt and
   Date.compare(date, end_date) == :lt
 ```
 
-This is verbose and therefore a little hard to read.
-It's also potentially incorrect, though not obviously so.
-What if `date` is considered "within" the interval even if it equals `start_date` or `end_date`?
-To include the bounds in our comparison, we'd instead write the expression like this:
+The above code is verbose, somewhat hard to read, and potentially incorrect (though not obviously so). What if `date` is considered "within" the interval inclusive of the `start_date` or `end_date`? To include the bounds in the comparison, you'd instead write the expression like this:
 
 ```elixir
 Date.compare(start_date, date) != :gt and
   Date.compare(date, end_date) != :gt
+
+# …or, terser but less performant:
+Date.compare(start_date, date) in [:lt, :eq]
 ```
 
-(We could have written `Date.compare(start_date, date) in [:lt, :eq]`, but `!= :gt` is faster.)
+To spot the difference between these two cases, you must keep in mind:
 
-In order to spot the difference between these two cases, you have to keep several things in mind:
+- the order of the arguments passed to `Date.compare/2`,
+- the specific comparison operators for each clause (`==` vs. `!=`), and
+- the specific comparison atoms for each clause (`:lt` vs. `:gt`).
 
-  * The order of the arguments passed to `Date.compare/2`
-  * The specific comparison operators for each clause: `==` vs. `!=`
-  * The specific comparison atoms for each clause: `:lt` vs. `:gt`
+Contrast this example with how equivalent Python code:
 
-Since this is hard to read, it's easy to introduce bugs.
-Contrast this with how you'd write the equivalent code in Python:
+```python
```

Comment:
**suggestion**: Drop the `python` language hint here.

It makes sense on GitHub:

<img width="345" alt="Screen Shot 2024-06-14 at 10 13 48 AM" src="https://github.com/CargoSense/compare_chain/assets/10274508/1f31e8a4-736b-4b21-ae7d-97ef7ac3fef9">

But it breaks on HexDocs:

<img width="296" alt="Screen Shot 2024-06-14 at 10 07 41 AM" src="https://github.com/CargoSense/compare_chain/assets/10274508/c5e0b64d-19f4-4752-b862-8a022c6e15d2">

vs. HexDocs today:

<img width="430" alt="Screen Shot 2024-06-14 at 10 08 24 AM" src="https://github.com/CargoSense/compare_chain/assets/10274508/1345ab2e-bb26-48f9-87cb-742b4272075f">

The GitHub vs. HexDocs formatting battle is a fun one :)

---

Sidenote: Amusingly, I can't make a code suggestion because GitHub doesn't understand that I want the code changed to exactly:

    ```

Reply:
I'm not sure I see what's broken here. Can you clarify?

Reply:
Syntax highlighting. Note how on HexDocs, the comments and operators aren't highlighted since ExDoc doesn't support:

    ```python

(Or at least I couldn't get it to.) I originally kept it plaintext since in this particular example, the Elixir and Python are equivalent in terms of syntax highlighting and ExDoc interprets:

    ```

as being Elixir.

Essentially, we can only get working highlighting in either GitHub or HexDocs, but not both.

Upon reflection though, I'm now leaning towards your change. Having Python syntax highlighting broken on HexDocs makes more sense than having it be misinterpreted in both places while also being broken on GitHub.

Reply:
> Upon reflection though, I'm now leaning towards your change. Having Python syntax highlighting broken on HexDocs makes more sense than having it be misinterpreted in both places while also being broken on GitHub.

I agree with this approach given the minimal side effect. The prose itself indicates that the following snippet is Python code, so that's helpful, and I'd be hesitant to take advantage of GitHub's highlighting since the document exists _on_ GitHub.

FWIW, npm does the same kind of README parsing thing as HexDocs and there are features they don't support, too, and it drives me slightly bananas.